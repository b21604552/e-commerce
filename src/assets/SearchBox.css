.search-wrapper {
    width: 695px;
    height: 48px;
    margin-right: 100px;
    background: #EEEEEE;
    border-radius: 100px;
    font-style: normal;
    font-weight: normal;
    font-size: 15px;
    line-height: 0;
    color: #B0B0B0;
    border: 0;
    display: flex;
    align-items: center;
    padding-left: 0.5rem;
    overflow: hidden;
    font-family: sans-serif;
}

.search-input {
    background: #EEEEEE;
    width: 643px;
    height: 48px;
    border: 0;
    font-style: normal;
    font-weight: 400;
    font-size: 15px;
    line-height: 0;
    color: #B0B0B0;
}

.search-input:focus {
    outline: none;
}

.prefix {
    width: 17.5px;
    height: 17.5px;
    margin-left: 27px;
    margin-right: 11.5px;
}