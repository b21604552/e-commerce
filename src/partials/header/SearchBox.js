import React from 'react'
import icon from '../../images/search-icon.svg'
import { setSearchedKeyword } from '../../services/redux/actions/GeneralAction'

export default function SearchBox () {

  function search (input) {
    let result = null

    if (input.length > 1)
      result = input

    setSearchedKeyword(result)
  }

  return (
    <div className={'search-wrapper'}>
      <span className="prefix"><img alt={'search-icon'} src={icon}/></span>
      <input id={'search-input'} type="text" className={'search-input'} placeholder={'25 milyon’dan fazla ürün içerisinde ara'} onChange={(e) => search(e.target.value)}/>
    </div>
  )
}