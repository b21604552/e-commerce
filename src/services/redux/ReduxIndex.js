import { createStore } from 'redux'
import reducerCombined from './reducerCombined'

export const reduxStore = createStore(reducerCombined)
