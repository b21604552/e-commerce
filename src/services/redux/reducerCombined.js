import { combineReducers } from 'redux'
import GeneralReducer from './reducers/GeneralReducer'

const reducerCombined = combineReducers({
  GeneralReducer: GeneralReducer
})

export default reducerCombined
