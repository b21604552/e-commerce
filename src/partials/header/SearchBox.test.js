import React from 'react'
import ReactDOM from 'react-dom'
import SearchBox from './SearchBox'
import renderer from 'react-test-renderer'
import { reduxStore } from '../../services/redux/ReduxIndex'
import { Provider } from 'react-redux'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

configure({ adapter: new Adapter() })

describe('SearchBox Test', () => {
  let page = (
    <Provider store={reduxStore}>
      <SearchBox/>
    </Provider>
  )

  it('Renders SearchBox without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(page, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders SearchBox correctly', () => {
    const div = renderer.create(page).toJSON()
    expect(div).toMatchSnapshot()
  })

  it('SearchBox Change Test 1', () => {
    const event = {
      preventDefault () {},
      target: { value: 'App' }
    }
    shallow(page).find(SearchBox).dive().find('#search-input').simulate('change', event)
    const { searchKeyword } = reduxStore.getState().GeneralReducer
    expect(searchKeyword).toStrictEqual('App')
  })

  it('SearchBox Change Test 2', () => {
    const event = {
      preventDefault () {},
      target: { value: 'A' }
    }
    shallow(page).find(SearchBox).dive().find('#search-input').simulate('change', event)
    const { searchKeyword } = reduxStore.getState().GeneralReducer
    expect(searchKeyword).toStrictEqual(null)
  })
})