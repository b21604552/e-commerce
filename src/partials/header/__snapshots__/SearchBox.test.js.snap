// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Test Renders SearchBox correctly 1`] = `
<div
  className="search-wrapper"
>
  <span
    className="prefix"
  >
    <img
      alt="search-icon"
      src="search-icon.svg"
    />
  </span>
  <input
    className="search-input"
    id="search-input"
    onChange={[Function]}
    placeholder="25 milyon’dan fazla ürün içerisinde ara"
    type="text"
  />
</div>
`;
